src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -1,281 +1,338 @@
-                                    throw new PaymentMethodFailedError(stripeResponse.error && stripeResponse.error.message);
-                                }
-
-                                const paymentPayload = {
-                                    methodId: payment.methodId,
-                                    paymentData: {
-                                        nonce: stripeResponse.paymentIntent.id,
-                                        shouldSaveInstrument,
-                                    },
-                                };
-
-                                return this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
-                            });
-                    });
-            });
+        return await this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload))
+            .catch(error => this._processAdditionalAction(error));
     }
 
     finalize(): Promise<InternalCheckoutSelectors> {
         return Promise.reject(new OrderFinalizationNotRequiredError());
     }
 
     deinitialize(): Promise<InternalCheckoutSelectors> {
-        if (this._cardElement) {
-            this._cardElement.unmount();
-        }
+        this._unmountElement();
 
         return Promise.resolve(this._store.getState());
     }
 
-    private _getStripeJs(): StripeV3Client {
-        if (!this._stripeV3Client) {
-            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+    private async _processAdditionalAction(error: unknown): Promise<any> {
+        if (!(error instanceof RequestError) || !some(error.body.errors, {code: 'additional_action_required'})) {
+            return Promise.reject(error);
         }
 
-        return this._stripeV3Client;
+        const action = error.body.additional_action_required;
+        if (action && action.type === 'redirect_to_url') {
+            return new Promise(() => {
+                window.location.replace(action.data.redirect_url);
+            });
+        }
     }
 
-    private _mapStripeBillingAddress(billingAddress: BillingAddress | undefined): StripeAddress | undefined {
-        if (!billingAddress) {
-            return undefined;
+    private async _confirmStripePayment(
+        paymentMethod: PaymentMethod,
+        shouldSaveInstrument: boolean = false
+    ): Promise<PaymentIntent> {
+        const { clientToken: clientSecret, returnUrl } = paymentMethod;
+
+        if (!clientSecret) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
         }
 
-        return {
-            city: billingAddress.city,
-            country: billingAddress.countryCode,
-            line1: billingAddress.address1,
-            line2: billingAddress.address2,
-            postal_code: billingAddress.postalCode,
-            state: billingAddress.stateOrProvinceCode,
-        };
+        let data;
+        let error: StripeError | undefined;
```

Comment:
I don't think you need `| undefined` for this and paymentIntent?

Reply:
It is required because confirm payment intent function returns one or the another but not both objects